dnf5-plugins/config-manager_plugin/unsetopt.cpp

```diff
@@ -107,48 +114,88 @@ void ConfigManagerUnsetOptCommand::configure() {
+                }
+            }
 
-        if (changed) {
-            parser.write(cfg_filepath, false);
+            if (changed) {
+                parser.write(cfg_filepath, false);
+            }
+        } else {
+            ctx.base.get_logger()->warning(
+                "config-manager: Request to remove main option but config file not found: {}", cfg_filepath.string());
         }
     }
 
-    auto repos_override_file_path = get_config_manager_repos_override_file_path(config);
-
     // Remove options from repositories overrides configuration file, remove empty sections.
-    if (!in_repos_opts_to_remove.empty() && std::filesystem::exists(repos_override_file_path)) {
-        ConfigParser parser;
-        bool changed = false;
-        parser.read(repos_override_file_path);
+    if (!in_repos_opts_to_remove.empty()) {
+        const auto & repos_override_file_path = get_config_manager_repos_override_file_path(config);
+        if (std::filesystem::exists(repos_override_file_path)) {
+            ConfigParser parser;
+            bool changed = false;
+            std::map<std::string, std::set<std::string>> used_repos_opts;
+
+            parser.read(repos_override_file_path);
+
+            std::vector<std::string> empty_config_sections;
+            for (const auto & [repo_id, setopts] : parser.get_data()) {
+                for (const auto & [in_repoid, keys] : in_repos_opts_to_remove) {
+                    if (sack::match_string(repo_id, sack::QueryCmp::GLOB, in_repoid)) {
+                        auto & used_repoid_opts = used_repos_opts[in_repoid];
+                        changed |= remove_from_config(parser, repo_id, keys, used_repoid_opts);
+                    }
+                }
+                if (setopts.empty()) {
+                    empty_config_sections.emplace_back(repo_id);
+                }
+            }
 
-        std::vector<std::string> empty_config_sections;
-        for (const auto & [repo_id, setopts] : parser.get_data()) {
+            // Generate warning for unused repoids and options
             for (const auto & [in_repoid, keys] : in_repos_opts_to_remove) {
-                if (sack::match_string(repo_id, sack::QueryCmp::GLOB, in_repoid)) {
-                    changed |= remove_from_config(parser, repo_id, keys);
+                if (const auto used_repoid_opts = used_repos_opts.find(in_repoid);
+                    used_repoid_opts == used_repos_opts.end()) {
+                    ctx.base.get_logger()->warning(
+                        "config-manager: Request to remove repository option but repoid not found in overrides: {}",
+                        in_repoid);
+                } else {
+                    for (const auto & key : keys) {
+                        if (!used_repoid_opts->second.contains(key))
+                            ctx.base.get_logger()->warning(
+                                "config-manager: Request to remove not present repository option from overrides: {}.{}",
```

Comment:
The same as the above. I suggest: "Request to remove repository option but it is not present in the overrides:"